CMake program target .elf path is wrong

Hi,
I am trying to setup a new modm project with CMake as build script generator. When calling the programming target with `make program` I get the following error:
```
** Programming Started **
Error: couldn't open PATH_TO_PROJECT_FOLDER/app/build/src/rot_encoder.elf
embedded:startup.tcl:1516: Error: ** Programming Failed **
in procedure 'modm_program' 
in procedure 'program' called at file "modm/openocd.cfg", line 13
in procedure 'program_error' called at file "embedded:startup.tcl", line 1581
at file "embedded:startup.tcl", line 1516
Built target program
```

I have the following file structure
```bash
├── app
│   ├── build
│   │   ├── rot_encoder.bin
│   │   ├── rot_encoder.elf
│   ├── CMakeLists.txt
│   ├── modm
│   ├── project.xml
│   ├── scripts
│   │   └── prog.cfg
│   └── src
│       └── main.cpp
└── ext
    └── modm
```

The problem comes from https://github.com/modm-io/modm/blob/ab7c1ca29a2bcc47e02ba1a991b6e138549d5303/tools/build_script_generator/cmake/resources/ModmConfiguration.cmake.in#L191
When I change the line to the following it just works:
```cmake
        ${PROJECT_BINARY_DIR}/${project_name}.elf   # removed src/
```

So my question is, am I doing something wrong? And why the .elf file is expected to be in a sub `build/src/` folder and not directly in the `build` folder?

I can reproduce this like so:

```sh
cd examples/stm32f469_discovery/assert/
lbuild -D ::build.path=build build
mkdir build
cd build
cmake ..
make -j4
make size
```
which then results in
```
 $ make size
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "modm/examples/stm32f469_discovery/assert/modm/modm_tools/size.py", line 162, in <module>
    output = format(args.source, eval(args.memories))
  File "modm/examples/stm32f469_discovery/assert/modm/modm_tools/size.py", line 79, in format
    with open(source, "rb") as src:
         ~~~~^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/niklas/development/modm-dev/modm/examples/stm32f469_discovery/assert/build/src/assert.elf'
make[3]: *** [size] Error 1
make[2]: *** [CMakeFiles/size.dir/all] Error 2
make[1]: *** [CMakeFiles/size.dir/rule] Error 2
make: *** [size] Error 2
FAIL
```

I think that's just a bug introduced by some major refactorings to this module at some point.
With your proposed fix, it works fine for me. Please open a PR with your fix so you get proper credit.

--- a/tools/build_script_generator/cmake/resources/ModmConfiguration.cmake.in
+++ b/tools/build_script_generator/cmake/resources/ModmConfiguration.cmake.in
@@ -181,35 +181,35 @@ function(modm_targets_create project_name)
   add_custom_target(size DEPENDS ${project_name}.elf)
   add_custom_command(TARGET size
     USES_TERMINAL
-    COMMAND cmake -E env PYTHONPATH=modm ${Python3_EXECUTABLE} -m modm_tools.size ${PROJECT_BINARY_DIR}/src/${project_name}.elf \"{{ memories }}\"
+    COMMAND cmake -E env PYTHONPATH=modm ${Python3_EXECUTABLE} -m modm_tools.size ${PROJECT_BINARY_DIR}/${project_name}.elf \"{{ memories }}\"
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
 
   add_custom_target(program DEPENDS ${project_name}.elf)
   add_custom_command(TARGET program
     USES_TERMINAL
     COMMAND cmake -E env PYTHONPATH=modm ${Python3_EXECUTABLE} -m modm_tools.openocd -f modm/openocd.cfg
-        ${PROJECT_BINARY_DIR}/src/${project_name}.elf
+        ${PROJECT_BINARY_DIR}/${project_name}.elf
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
 
   add_custom_target(program-bmp DEPENDS ${project_name}.elf)
   add_custom_command(TARGET program-bmp
     USES_TERMINAL
     COMMAND cmake -E env PYTHONPATH=modm ${Python3_EXECUTABLE} -m modm_tools.bmp -p ${MODM_BMP_PORT}
-        ${PROJECT_BINARY_DIR}/src/${project_name}.elf
+        ${PROJECT_BINARY_DIR}/${project_name}.elf
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
 
   add_custom_target(program-jlink DEPENDS ${project_name}.elf)
   add_custom_command(TARGET program-jlink
     USES_TERMINAL
     COMMAND cmake -E env PYTHONPATH=modm ${Python3_EXECUTABLE} -m modm_tools.jlink -device {{ jlink_partname }}
-        ${PROJECT_BINARY_DIR}/src/${project_name}.elf
+        ${PROJECT_BINARY_DIR}/${project_name}.elf
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
 
   add_custom_target(debug DEPENDS ${project_name}.elf)
   add_custom_command(TARGET debug
     USES_TERMINAL
     COMMAND cmake -E env PYTHONPATH=modm ${Python3_EXECUTABLE} -m modm_tools.gdb -x modm/gdbinit -x modm/openocd_gdbinit
-        --elf ${PROJECT_BINARY_DIR}/src/${project_name}.elf --ui=${MODM_DBG_UI}
+        --elf ${PROJECT_BINARY_DIR}/${project_name}.elf --ui=${MODM_DBG_UI}
         openocd -f modm/openocd.cfg
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
 
@@ -217,7 +217,7 @@ function(modm_targets_create project_name)
   add_custom_command(TARGET debug-bmp
     USES_TERMINAL
     COMMAND cmake -E env PYTHONPATH=modm ${Python3_EXECUTABLE} -m modm_tools.gdb -x modm/gdbinit -x modm/openocd_bmp
-        --elf ${PROJECT_BINARY_DIR}/src/${project_name}.elf --ui=${MODM_DBG_UI}
+        --elf ${PROJECT_BINARY_DIR}/${project_name}.elf --ui=${MODM_DBG_UI}
         bmp -p ${MODM_BMP_PORT}
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
 
@@ -225,7 +225,7 @@ function(modm_targets_create project_name)
   add_custom_command(TARGET debug-jlink
     USES_TERMINAL
     COMMAND cmake -E env PYTHONPATH=modm ${Python3_EXECUTABLE} -m modm_tools.gdb -x modm/gdbinit -x modm/openocd_jlink
-        --elf ${PROJECT_BINARY_DIR}/src/${project_name}.elf --ui=${MODM_DBG_UI}
+        --elf ${PROJECT_BINARY_DIR}/${project_name}.elf --ui=${MODM_DBG_UI}
         jlink -device {{ jlink_partname }}
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
 
@@ -233,7 +233,7 @@ function(modm_targets_create project_name)
   add_custom_command(TARGET debug-coredump
     USES_TERMINAL
     COMMAND cmake -E env	PYTHONPATH=modm ${Python3_EXECUTABLE} -m modm_tools.gdb -x modm/gdbinit
-        --elf ${PROJECT_BINARY_DIR}/src/${project_name}.elf --ui=${MODM_DBG_UI}
+        --elf ${PROJECT_BINARY_DIR}/${project_name}.elf --ui=${MODM_DBG_UI}
         crashdebug
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
 

